pkg/repository/worker_redis.go

```diff
@@ -310,18 +310,19 @@ func (r *WorkerRedisRepository) UpdateWorkerCapacity(worker *types.Worker, reque
 	key := common.RedisKeys.SchedulerWorkerState(worker.Id)
 
 	// Retrieve current worker capacity
-	w, err := r.getWorkerFromKey(key)
+	currentWorker, err := r.getWorkerFromKey(key)
 	if err != nil {
 		return fmt.Errorf("failed to get worker state <%v>: %v", key, err)
 	}
 
+	sourceWorker := currentWorker // worker from the Redis store
+	if sourceWorker == nil {
+		sourceWorker = worker // worker from the argument
+	}
+
```

Comment:
I think you can just declare sourceWorker in the assignment such that L318 isn't needed

Reply:
We did something in v1 where this logic needed to be this way. Could have been an oversight, but I don't think I want to rock the boat on this too much. I just wanted to avoid having to write and return in CopyStruct twice.